Sources/Misc/Concurrency/OperationDispatcher.swift

```diff
@@ -43,13 +42,29 @@ class OperationDispatcher {
 
     func dispatchOnWorkerThread(withRandomDelay: Bool = false, block: @escaping @Sendable () -> Void) {
         if withRandomDelay {
-            let delay = Double.random(in: 0..<self.maxJitterInSeconds)
-            self.workerQueue.asyncAfter(deadline: .now() + delay, execute: block)
+            self.workerQueue.asyncAfter(deadline: .now() + Self.randomDelay(), execute: block)
         } else {
             self.workerQueue.async(execute: block)
         }
     }
 
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    func dispatchOnWorkerThread(withRandomDelay: Bool = false, block: @escaping @Sendable () async -> Void) {
+        Task.detached(priority: .background) {
+            if withRandomDelay {
+                try? await Task.sleep(nanoseconds: DispatchTimeInterval(Self.randomDelay()).nanoseconds)
+            }
+
+            await block()
+        }
+    }
+
+    private static func randomDelay() -> TimeInterval {
+        Double.random(in: 0..<Self.maxJitterInSeconds)
+    }
```

Comment:
Can we add a comment for the random delay? I'm not actually sure why its here and I probably won't remember when I come back and look at this again 😛

Reply:
👍🏻  (BTW this isn't new, just moving code)

Reply:
Done